fontforgeexe/cvpalettes.c

```diff
@@ -3790,7 +3790,19 @@ void _CVPaletteActivate(CharView *cv,int force) {
 	    else
 		SaveOffsets(old->gw,cvlayers,&cvlayersoff);
 	}
-	GDrawSetUserData(cvtools,cv);
+
```

Comment:
I'm not thrilled with the two calls to `CVPalletteCheck()` (given that the one at the start of the function remains) and the potential for creating, destroying, and creating the palette windows when `force` is true. 

But beyond that pickiness, is the logic right? I presume that `GDrawDestroyWindow()` will throw away the visibility statuses set starting on line 3772. Is that OK? If it is, can't we move the `if ( (old = GDrawGetUserData(cvtools))!=cv || force) {` to the top and put that stuff in an else, eliminating the double `CVPaletteCheck()`? And if it isn't OK, doesn't the code need a broader reorganization?  

(Also, the indentation is wonky in this version.)

Reply:
Discarding current visibility state is probably ok, since the rest of the activation routines determine what should be visible. As for removing the topmost `CVPaletteCheck`, yeah it's probably possible, but I'd have to study this more to find out the consequences - and that's a problem for another day (or patches welcome).